lib/pbench/server/api/auth.py

```diff
@@ -79,6 +79,27 @@ def get_secret_key(self):
         except Exception as e:
             Auth.logger.exception(f"{__name__}: ERROR: {e.__traceback__}")
 
+    def verify_user(self, target_username: str) -> User:
+        """
+        Check whether the requested target user is the owner of the authorization
+        token provided to the API.
+
+        We're returning a User instance corresponding to the target username provided.
+        If the request is not authenticated (current user is None) we return None as
+        we won't be able to verify the target user. However, if the current user is
+        an admin, we return the instance of the target user by querying the target
+        username provided.
```

Comment:
I think Webb's take above would be an improvement on this rather dense paragraph! But I'm not going to gate this PR on any improvement here.